src/encoded/audit/experiment.py

```diff
@@ -4042,6 +4042,41 @@ def audit_experiment_inconsistent_analyses_files(value, system, files_structure)
         yield AuditFailure('inconsistent analyses files', detail, level='INTERNAL_ACTION')
 
 
+def audit_experiment_inconsistent_genetic_modifications(value, system, excluded_types):
+    genetic_modifications = {'no genetic modifications': set()}
+
+    if value['status'] in ['deleted', 'replaced', 'revoked']:
+        return
+
+    if value['assay_term_name'] == 'pooled clone sequencing':
+        return
+
+    if value.get('replicates') is not None and len(value['replicates']) > 1:
+        for rep in value['replicates']:
+            if (rep['status'] not in excluded_types and 'library' in rep and rep['library']['status'] not in excluded_types and 'biosample' in rep['library'] and rep['library']['biosample']['status'] not in excluded_types):
+                biosampleObject = rep['library']['biosample']
+                modifications = biosampleObject.get('applied_modifications')
+                if not modifications:
+                    genetic_modifications['no genetic modifications'].add(biosampleObject['@id'])
+                else:
+                    for gm in modifications:
+                        gm_ids = set()
+                        gm_ids.add(gm['@id'])
+                        gm_combined = tuple(sorted(gm_ids))
```

Comment:
Not sure if things are right here. For what you have here, doesn't that end up just `gm_combined = (modifications[-1]['@id],)`?

Do you actually want `gm_combined = tuple(sorted(gm['@id'] for gm in modifications))`?

Reply:
Ah yes you're right, my mistake. Comprehension is better too, will push a commit to change it.